### PR TITLE
Streetcode art&text unit tests

### PR DESCRIPTION
**Pull request by Chynchenko**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: Streetcode/Streetcode.XUnitTest/MediatRTests/MediaTests/StreetcodeArtTests/GetStreetcodeArtByStreetcodeIdTest.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                    Streetcode = null,
+                    ArtId = 2,
+                    Art=null
+                }
+            };
+        }
+
+        [Theory]
+        [InlineData(1)]
+        public async Task GetStreetcodeArtByStreetcodeId_ReturnsSuccesfullyStreetcodeArt(int streetcodeId)
```
Comment (Katerix) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/MediaTests/StreetcodeArtTests/GetStreetcodeArtByStreetcodeIdTest.cs:74]: move setup's to private method's. don't repeat yourself)

File: Streetcode/Streetcode.XUnitTest/MediatRTests/MediaTests/StreetcodeArtTests/GetStreetcodeArtByStreetcodeIdTest.cs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        {
+            repository.Setup(r => r.StreetcodeArtRepository.GetAllAsync(It.IsAny<Expression<Func<StreetcodeArt, bool>>>(),
+                It.IsAny<Func<IQueryable<StreetcodeArt>, IIncludableQueryable<StreetcodeArt, object>>>()))
+                .ReturnsAsync(GetStreetcodeArtsList());
+
+            mockMapper.Setup(x => x.Map<IEnumerable<StreetcodeArtDTO>>(It.IsAny<IEnumerable<object>>())).Returns(GetStreetcodeArtDTOList());
```
Comment (Katerix) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/MediaTests/StreetcodeArtTests/GetStreetcodeArtByStreetcodeIdTest.cs:80]: this one too.

File: Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/DeleteTextTest.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+            repository.Setup(x => x.SaveChangesAsync()).ReturnsAsync(1);
+
+            var handler = new DeleteTextHandler(repository.Object);
+
+            var result = await handler.Handle(new DeleteTextCommand(id), CancellationToken.None);
+
+            Assert.Multiple(
+            () => Assert.NotNull(result),
+            () => Assert.True(result.IsSuccess));
```
Comment (Katerix) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/DeleteTextTest.cs:51]: As for me these checks are useless. If result is not null (first assert) that the result is obviously successful. Maybe it is better to check the type of result.Value. You can also check if the element was really deleted

File: Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/DeleteTextTest.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        }
+
```
Comment (Katerix) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/DeleteTextTest.cs:53]: delete empty row

File: Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/GetAllTextsTest.cs [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            new TextDTO() { Id = 1 }
+        };
+
+            var testTexts = new Text() { Id = 1 };
+
+            repository.Setup(repo => repo.TextRepository.GetFirstOrDefaultAsync(It.IsAny<Expression<Func<Text, bool>>>(), It.IsAny<Func<IQueryable<Text>, IIncludableQueryable<Text, Text>>?>()))
+                .ReturnsAsync(testTexts);
+
+            repository.Setup(repo => repo.TextRepository.GetAllAsync(null, null)).ReturnsAsync(testTextsList);
+
```
Comment (Katerix) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/GetAllTextsTest.cs:44]: why two repository setups?? the last hides the first one. Also move them to private methods)

File: Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/GetAllTextsTest.cs [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+            var mockMapper = new Mock<IMapper>();
+
+
+                repository.Setup(repo => repo.TextRepository.GetFirstOrDefaultAsync(It.IsAny<Expression<Func<Text, bool>>>(),
+                It.IsAny<Func<IQueryable<Text>, IIncludableQueryable<Text, Text>>?>()))
+                .ReturnsAsync((Text)null);
+
+            repository.Setup(repo => repo.TextRepository.GetAllAsync(null, null)).ReturnsAsync((List<Text>)null);
+
+            mockMapper.Setup(x => x.Map<IEnumerable<TextDTO>>(It.IsAny<IEnumerable<object>>())).Returns(new List<TextDTO>() { new TextDTO() { Id = 1 } });
```
Comment (Katerix) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/GetAllTextsTest.cs:69]: the row 69 is very long. You can break it into smaller rows of code

File: Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/GetAllTextsTest.cs [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+
```
Comment (Katerix) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/GetAllTextsTest.cs:71]: you can also delete extra space. You don't need it here

File: Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/GetTextByIdTest.cs [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+            {
+                return new TextDTO { Id = sourceText.Id };
+            });
+
+            var handler = new GetTextByIdHandler(repository.Object, mockMapper.Object);
+
+            var result = await handler.Handle(new GetTextByIdQuery(2), CancellationToken.None);
+
+            Assert.NotNull(result);
+            Assert.True(result.IsSuccess);
```
Comment (Katerix) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/GetTextByIdTest.cs:74]: multiple asserts

**Review by Katerix — CHANGES_REQUESTED** [review]

File: Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/GetAllTextsTest.cs (hunk 7 above)
Comment (Chynchenko) [reply Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/GetAllTextsTest.cs:71]: done

File: Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/GetAllTextsTest.cs (hunk 6 above)
Comment (Chynchenko) [reply Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/GetAllTextsTest.cs:69]: done

File: Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/GetTextByIdTest.cs (hunk 8 above)
Comment (Chynchenko) [reply Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/GetTextByIdTest.cs:74]: not sure

**Pull request merged**